Update config docs (cleanup, packit_instances, issue_repository) (#421)
packit/packit.dev: content/docs/configuration.md

You can enable the commenting by setting `successful_build` to `true`.

#### issue_repository

Use this key to be notified about errors of the downstream jobs (Koji build and Bodhi update).

> This repository can be any GitHub/GitLab/Pagure repository
> where issues are enabled and Packit has an identity on that instance.
> (Let us know if you need some other instance to be supported.)

Reviewer: ```suggestion
This repository can be any GitHub/GitLab/Pagure instance
where issues are enabled and Packit has an identity.
(Let us know if you need some other instances to be supported.)
```

Author: I might have mixed those two config options together, but here, we are really talking about repositories. As I am looking at that, I'll make it more clear and mention that it needs to be a URL.

Author: Can you please take another look if it is clear now?

Be aware that this is a preparation phase and we want to build all the packages in Copr by default.

#### packit_instances
(*list of strings*) If you want to help us with catching issues or need some feature sooner than other users,
you can use our staging instance that has the freshest code we have.
For that, you can specify the instance(s) that will react to your jobs by using this configuration option.
It uses `["prod"]` as the default, but you can set both (`["prod", "stg"]`) or just stage (`["stg"]`).
You can also have a different setup for each job -- see [how the overriding works](#overriding-global-parameters)
in the config file.

Just be aware that:
* You still need to install [the staging GitHub application](https://github.com/apps/packit-as-a-service-stg).
* Staging application has fewer resources and can be a bit slower in responses.
* If you use both instances, you will have twice the number of commit statuses.
* Picking just one instance might be better for downstream jobs since
  both instances will work with the production instances of Fedora systems.

Please, let us know when something looks weird or does not work with the staging app.

> By doing that, you are helping us be sure that we don't break your use-case.

Reviewer: ```suggestion
By doing that, you are helping us to be sure that we don't break your use-case.
```
